models/reservation.js

```diff
@@ -106,45 +63,34 @@ exports.reserve = function(driverId, latitude, longitude) {
       reservation.attributes.longitude = spotLatLong.longitude;
       reservation.attributes.hostId = spot.attributes.userId;
       reservation.attributes.spotId = spot.id;
-      return reservation;
-    })
-    .then(createNewReservation)
-    .then(function(reservation) {
-      return Spot.addReservationId(reservation.attributes.spotId, reservation.id);
+      return reservationsRef.child(reservation.id).update(reservation.attributes);
```

Comment:
Ya its pretty dumb. I thought it was just a promise issue but their callback for update only returns errors as well. Dumb that I need to follow up this promise w/ a request to exports.get